### PR TITLE
WIP: convert to TS

### PR DESCRIPTION
**Pull request by Pergamene**

### CODE REVIEW

File: src/markdown-to-inner-partitions.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
+import { PARTITION_TYPES } from './partition-types';
+
+export function generateInnerPartitions(markdownSubstring: string): object | null {
```
Comment (rhyeen) [inline src/markdown-to-inner-partitions.ts:3]: `object` should be `Partition[]` instead

Comment (rhyeen) [reply src/markdown-to-inner-partitions.ts:3]: Also, don't return `null`.  Either return an empty `Partition[]` or throw an error.

File: src/markdown-to-inner-partitions.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  if (markdownSubstring.length === 0) {
+    return null;
+  }
+  let partitions = [];
```
Comment (rhyeen) [inline src/markdown-to-inner-partitions.ts:7]: const

File: src/markdown-to-inner-partitions.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  let currentIndex = 0;
+  let nextIndex = _findNextInnerStart(currentIndex, markdownSubstring);
+  if (nextIndex === markdownSubstring.length) {
+    return null;
+  }
+  let currentChar: string, close: number;
```
Comment (rhyeen) [inline src/markdown-to-inner-partitions.ts:13]: on separate lines

File: src/markdown-to-inner-partitions.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    nextIndex = _findNextInnerStart(currentIndex, markdownSubstring);
+  }
+  if (currentIndex !== nextIndex) {
+    partitions.push(_buildTextPartition(markdownSubstring.substring(currentIndex)));
+  }
+  return partitions;
+}
+
+function _buildBoldPartition(markdownSubstring: string): object {
+  let partitions = generateInnerPartitions(markdownSubstring);  //@TODO: NEED TYPE? (see line 57)
```
Comment (rhyeen) [inline src/markdown-to-inner-partitions.ts:48]: const

File: src/markdown-to-inner-partitions.ts [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+    }
+    nextIndex = _findNextInnerStart(currentIndex, markdownSubstring);
+  }
+  if (currentIndex !== nextIndex) {
+    partitions.push(_buildTextPartition(markdownSubstring.substring(currentIndex)));
+  }
+  return partitions;
+}
+
+function _buildBoldPartition(markdownSubstring: string): object {
```
Comment (rhyeen) [inline src/markdown-to-inner-partitions.ts:47]: Fix this an all following returns to be `Partition` or whatever other class is actually returned.

File: src/markdown-to-inner-partitions.ts [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+  let partitions = generateInnerPartitions(markdownSubstring);  //@TODO: NEED TYPE? (see line 57)
+  if (!partitions) {
+    return { type: PARTITION_TYPES.BOLD, value: markdownSubstring.replace(/\\/g, '') };
```
Comment (rhyeen) [inline src/markdown-to-inner-partitions.ts:50]: Won't need `PARTITION_TYPES.*` once you have a `Partition` class and a `BoldPartition` class that extends it.

**Review by rhyeen — CHANGES_REQUESTED** [review]

File: src/markdown-to-inner-partitions.ts (hunk 1 above)
Comment (Pergamene) [reply src/markdown-to-inner-partitions.ts:3]: What is wrong with `null`?  I'll have to see if I'm checking for `null` in other places.

**Comment by Pergamene** [comment]
Copied in your file, but I removed all `PageDetail` from the namings since my project doesn't have the context of page details.  Is that going to screw things up in spiderweb?

**Pull request closed without merging**